### PR TITLE
[TS] Fix Sidebar, Stack, Tab and Tabs extended types

### PR DESCRIPTION
**Pull request by miguelmarcondesf**
#### What does this PR do?
Add ExtendedProps declarations for Sidebar, Stack, Tab and Tabs components, forming part of issue #4998.

#### Where should the reviewer start?
Reviewers can start from any of the files, there were independent changes only in TS declaration files

#### What testing has been done on this PR?
No additional tests were added

#### How should this be manually tested?
The import of any of the modified components in question will now have the respective _JSX.IntrinsicElements_ property 

#### Any background context you want to provide?
Discussed in the #4978

#### What are the relevant issues?
Progress track: #4998

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Yes

### CODE REVIEW

**Review by abnersajr — APPROVED** [review]

**Review by ShimiSun — COMMENTED** [review]
Overall it looks good thanks for the contribution.

One question though, for some components, we are exporting a type `export type TabsExtendedProps` and in some other cases we are exporting an interface, e.g. `export interface StackExtendedProps`

It might be a trivial question, but wouldn't it impact the user consumption of the type differently? Should we thrive to be more consistent in the way we export?
@abnersajr

**Comment by miguelmarcondesf** [comment]
Hey @ShimiSun 
I had the following problems trying to use interface on Tabs and Sidebar component, each one with a specific props conflict 
image
image

**Comment by abnersajr** [comment]
@ShimiSun, we need to use `export interface` always it is possible.

@miguelmarcondesf in your scenario you can do something like this:

https://github.com/grommet/grommet/pull/5021/files#diff-37524e012a5eb198c557ffc83a19486994866790935ef45e467d68096560ca90R26

Once we expect custom children, we can omit it from the `JSX.intrinsictElement` and if necessary add it to our prop.

File: src/js/components/Tabs/index.d.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ export interface TabsProps {
   onActive?: (index: number) => void;
 }
 
-declare const Tabs: React.FC<TabsProps & JSX.IntrinsicElements['div']>;
+export type TabsExtendedProps = TabsProps & JSX.IntrinsicElements['div'];
```
Comment (abnersajr) [inline src/js/components/Tabs/index.d.ts:23]: Use this as references to solve properties conflict: https://github.com/grommet/grommet/pull/5021/files#diff-37524e012a5eb198c557ffc83a19486994866790935ef45e467d68096560ca90R26

File: src/js/components/Sidebar/index.d.ts [hunk 2]
```diff
@@ -6,8 +6,10 @@ export interface SidebarProps {
   header?: React.ReactNode;
 }
 
-declare const Sidebar: React.FC<BoxProps &
+export type SidebarExtendedProps = BoxProps &
```
Comment (abnersajr) [inline src/js/components/Sidebar/index.d.ts:9]: Use this as references to solve properties conflict: https://github.com/grommet/grommet/pull/5021/files#diff-37524e012a5eb198c557ffc83a19486994866790935ef45e467d68096560ca90R26

**Review by abnersajr — CHANGES_REQUESTED** [review]

**Comment by miguelmarcondesf** [comment]
Hi @ShimiSun and @abnersajr 
I followed the suggestions, now I believe everything is right.

**Review by abnersajr — APPROVED** [review]

File: src/js/components/Sidebar/index.d.ts [hunk 3]
```diff
@@ -6,9 +6,12 @@ export interface SidebarProps {
   header?: React.ReactNode;
 }
 
-export type SidebarExtendedProps = BoxProps &
-  SidebarProps &
-  JSX.IntrinsicElements['div'];
+type divProps = Omit<JSX.IntrinsicElements['div'], 'onClick'>;
```
[the remark this answers is not in the document]
Comment (ShimiSun) [reply src/js/components/Sidebar/index.d.ts:9]: Which IDEA addon are you using to see this information?

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**